source/functions/BEMIO/waveNumber.m

```diff
@@ -1,125 +1,42 @@
-%   dispersion, Coastal Engineering, 46(2), 71-74.
-% A simple correction is applied to the initial guess to account for
-% for currents.
-%
-% Input: 
-%   wa = absolute angular frequency [rad/T]
-%   wd = wave direction, going to [rad]
-%   h = water depth [L]
-%   u = current velocity in x direction [L/T]
-%   v = current velocity in y direction [L/T]
-%   tol = wave number error tolerance for Newton-Raphson method 
-%         (default is 0.0001 rad/L)
-%
-% Output:
-%   wk = wave number [rad/L]
-%   err = final error defined as g*k*tanh(k*h) -sig^2 [rad^2/T^2]
-%         (always smaller than tol)
-%   iter = number of iterations 
-%
-% Author: Alex Sanchez, USACE-CHL
-% Date: February 27, 2012
-% Version 2.0
-
-twopi = 2*pi; %Internal variable
-
-%Demo mode
-% if nargin==0
-    g = 9.81; %gravity [m/s^2]
-%     T = 5:10; %wave period [s]
-%     freq = 1./T; %frequency [1/s]
-%     wa = twopi*freq; %absolute angular frequency [rad/s]
-    wd = 0*pi/180; %wave angle [rad]
-%     h = 6; %water depth [m]
-    u = 0;  %current velocity in x direction [m/s]
-    v = 0;  %current velocity in y direction [m/s]
-% end
-
-%Make initial guess with Guo (2002)
-xi = wa./sqrt(g./h); %note: =h*wa/sqrt(h*g/h)
-yi = xi.*xi/(1.0-exp(-xi.^2.4908)).^0.4015;
-wki = yi/h; %Initial guess without current-wave interaction
-
-%Current velocity in wave direction
-uk = cos(wd).*u + sin(wd).*v; 
-
-%Simple correction for currents (tends to overcorrect, hense the relaxation)
-wki = 0.3*wki + 0.7*(wa-wki.*uk).^2./(g*tanh(wki.*h));
-
-%Save wavenumber for iteration
-wk = wki; 
-
-%Save for screen output
-if nargin==0
-    wleni = twopi./wki; %initial wave length [m]
+function k = waveNumber(w,waterDepth,g,deepWaterWave)
+% Solves the wave dispersion relation w^2 = g*k*tanh(k*h) for the wave
+% number
+% 
+% Parameters
+% ----------
+%     w : [1 1] float
+%         Wave frequency [rad/s]
+%     
+%     waterDepth : [1 1] float
+%         Water depth [m]
+%     
+%     g : [1 1] float
+%         Gravitational acceleration [m/s^2]
+%     
+%     deepWaterWave : [1 1] float
+%         waveClass flag inidicating a deep water wave [-]
+% 
+% Returns
+% ------------
+%    k : [1 1] float
+%        Wave number [m]
+% 
+
+arguments
+    w
+    waterDepth
+    g = 9.81
+    deepWaterWave = 0;
 end
 
-%Set default tolerance
-if nargin<7
-    tol = 0.0001; %default tolerance for Newton-Raphson iteration loop
-end
+% Initial guess
+k = w.^2./g;
 
-%Main loop for Newton-Raphson iteration
-for iter=1:10
-    sig = wa-wk.*uk; 
-    tanhkh = tanh(wk.*h);
-    f = g*wk.*tanhkh - sig.^2;
-    fp = g*(tanhkh - h.*wk.*(tanhkh.^2-1)) + 2*uk.*sig;
-    wk = wki - f./fp;
-    iblk = (wk<0 | wk>1e6); %No solution, assume wave blocking
-    wk(iblk) = NaN;
-    wkchg = abs(wk-wki);
-    if(all(iblk) || all(wkchg<tol)), break, end
-    wki = wk;
+% Iterate for shallow and intermediate depths
+if deepWaterWave == 0
+    for i = 1:100
+        k = w.^2./g./tanh(k.*waterDepth);
+    end
 end
 
```

Comment:
Note to self to double check this formulation for shallow, intermediate and deep water

Reply:
Verified this is correct, and added some comments to clarify